src/main/groovy/com/netflix/nebula/lint/plugin/GradleLintPluginTaskConfigurer.groovy

```diff
@@ -140,14 +140,20 @@ class GradleLintPluginTaskConfigurer extends AbstractLintPluginTaskConfigurer {
      * @param lintTasks
      * @param autoLintTask
      */
-    private void finalizeAllTasksWithAutoLint(Project project, List<TaskProvider> lintTasks, Task autoLintTask) {
+    private void finalizeAllTasksWithAutoLint(Project project, List<TaskProvider> lintTasks, Task autoLintTask, GradleLintExtension lintExt) {
+        boolean skipForSpecificTask = project.gradle.startParameter.taskNames.any { lintExt.skipForTasks.contains(it) }
+
+        if(skipForSpecificTask) {
+            return
+        }
+
         project.tasks.configureEach { task ->
-            if (!lintTasks.contains(task) && !task.name.contains(AUTO_LINT_GRADLE) && !task.name.contains(CLEAN_TASK_NAME) && !task.name.contains(DEPENDENCY_INSIGHT_TASK_NAME)) {
+            if (!lintTasks.contains(task) && !task.name.contains(AUTO_LINT_GRADLE) && !task.name.contains(CLEAN_TASK_NAME)) {
```

Comment:
kept the check on clean task name because if we follow `project.gradle.startParameter.taskNames.any` pattern, it would discard builds that use `./gradlew clean build`